modules/rte_mis_demo_content/src/TestData.php

```diff
@@ -0,0 +1,524 @@
+      // Get the type of area from sixth column.
+      $district = $sheetData->getCell([6, $rowNumber])->getValue();
+      // Get the type of area from seventh column.
+      $block = $sheetData->getCell([7, $rowNumber])->getValue();
+      // Validate parameter before creating school udise code.
+      $validAidStatus = SchoolBatch::getValidListValue($aidStatus, 'field_aid_status');
+      $validMinorityStatus = SchoolBatch::getValidListValue($minorityStatus, 'field_minority_status');
+      $validTypeOfArea = SchoolBatch::getValidListValue($typeOfArea, 'field_type_of_area');
+      $blockTid = SchoolBatch::getBlockIdLocation($district, $block);
+      $userId = $this->currentUser->id();
+      if (!empty(trim($udiseCode)) && !empty(trim($schoolName)) && is_numeric($udiseCode) && strlen($udiseCode) === 11
+      && $validAidStatus && $validTypeOfArea && $validMinorityStatus && $blockTid) {
+        $existingTerm = NULL;
+        if (empty($existingTerm)) {
+          // Create new UDISE code if it does not exist.
+          // Also store user ip, mark upload_type as `bulk_upload` and
+          // set workflow status to `approved`.
+          $term = Term::create([
+            'name' => trim($udiseCode),
+            'field_school_name' => trim($schoolName),
+            'vid' => 'school',
+            'field_workflow' => 'school_workflow_approved',
+            'field_upload_type' => 'bulk_upload',
+            'field_minority_status' => $validMinorityStatus,
+            'field_type_of_area' => $validTypeOfArea,
+            'field_aid_status' => $validAidStatus,
+            'field_location' => $blockTid,
+          ]);
+          $term->setRevisionUser($this->entityTypeManager->getStorage('user')->load($userId));
+          $term->save();
+        }
+      }
+      // Update our progress information.
+      $context['sandbox']['progress']++;
+      $context['message'] = $this->t(
+        'Completed @current out of @max',
+        [
+          '@current' => $context['sandbox']['progress'],
+          '@max' => $context['sandbox']['max'],
+        ]
+      );
+    }
+    // Inform the batch engine that we are not finished,
+    // and provide an estimation of the completion level we reached.
+    if ($context['sandbox']['progress'] != $context['sandbox']['max']) {
+      $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
+    }
+  }
+
+  /**
+   * Create Academic Sessions.
+   */
+  public function createAcademicSession() {
+    $storage = $this->entityTypeManager->getStorage('mini_node');
+    // Check for existing active campaign before creating one.
+    $existing_active_campaign = $storage->loadByProperties([
+      'type' => 'academic_session',
+      'field_academic_year' => '2024_25',
+      'status' => 1,
+    ]);
+    if (empty($existing_active_campaign)) {
+      $node = $storage->create([
+        'type' => 'academic_session"',
```

Comment:
Please check here, this will lead to fatal error.

Reply:
fixed.